core/space/services/services_sharing.go

```diff
@@ -0,0 +1,91 @@
+package services
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+
+	"github.com/FleekHQ/space-daemon/core/space/domain"
+)
+
+func (s *Space) GenerateFileSharingLink(
+	ctx context.Context,
+	path string,
+	bucketName string,
+) (*domain.FileSharingInfo, error) {
+	key, err := s.keychain.GenerateTempKey()
+	if err != nil {
+		return nil, err
+	}
+
+	bucket, err := s.getBucketWithFallback(ctx, bucketName)
+	if err != nil {
+		return nil, err
+	}
+
+	encryptedFile, err := s.createTempFileForPath(ctx, path)
+	if err != nil {
+		return nil, err
+	}
+	defer encryptedFile.Close()
+
+	writer, err := newEncryptedFileWriter(encryptedFile, key)
+	if err != nil {
+		return nil, err
+	}
+
+	err = bucket.GetFile(ctx, path, writer)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Publish file to ipfs
+
+	return &domain.FileSharingInfo{
+		Bucket:         bucketName,
+		Path:           path,
+		SharedFileHash: "",
+		SharedFileKey:  writer.EncodeKey(),
+		Link:           "https://space.storage/shared/{hash}/{decryption_key}",
```

Comment:
@jsonsivar @dmerrill6 What should be the format of the link we would be sharing? This is currently the format I have in mind. Ideally, the link should map to a server we have access to, so we can fetch the hash from ipfs and decrypt with the key provided.

Reply:
I was thinking if we upload the file along with an index.html both in a folder to IPFS, then any gateway can serve up the decrypted file without depending on a backend server.

Reply:
Yeah, I'm doing some tests to know what the content of the index.html would be, so as to decrypt the file and display it.

Reply:
Nice! On that note I suppose the decryption key would be a query string parameter in the link here. 

Also regarding the publishing to IPFS, was looking at some examples so just pasting some snippet here:
```
import httpapi "github.com/ipfs/go-ipfs-http-client"
IpfsAddr = tb.cfg.GetString(config.Ipfsaddr, "/ip4/127.0.0.1/tcp/5001")

ic, err := httpapi.NewApi(conf.AddrIPFSAPI)
if err != nil {
  return nil, err
}

pth, err := ic.Unixfs().Add(
  ctx,
  ipfsfiles.NewReaderFile(r),
  options.Unixfs.CidVersion(1),
  options.Unixfs.Pin(true))
if err != nil {
  return nil, err
}

return ic.ResolveNode(ctx, pth)
```